src/DetectorSD.cpp

```diff
@@ -34,7 +34,20 @@ G4bool DetectorSD::ProcessHits(G4Step* aStep, G4TouchableHistory* )
 {
     G4double edep = aStep->GetTotalEnergyDeposit();
 
-    if(edep==0.0) return false;
+    if(edep==0.0){
+      if(aStep->GetPostStepPoint()->GetMomentum().mag2() > 0){
+      // particle quanta interact in detector but does not deposit energy
```

Comment:
Just a question:L Is it possible to check at this level explicitely some Geant code/flag that points to Rayleigh scattering?

Reply:
Yes, you can. Ask for aStep->GetPostStepPoint()->GetProcessDefinedStep()->GetProcessName()

Reply:
So maybe you could use it in addition to this condition to be sure that it is Railegh?
It would be more expressive.

Reply:
I mean to check the process name if(...=="Reighlei) ...

Reply:
Then we need to take into account all types of elastic scatterings (which depends on Physics List) and there always will be possibility that we will miss something and discover CPT violation.
Therefore I would rather stick to solution  if ( deposit energy) then hit else { if (changing momentum) put flag}